core/domain/calculation_registry_test.py

```diff
@@ -0,0 +1,54 @@
+# coding: utf-8
+#
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for calculation registry."""
+
+import inspect
+import random
+
+from core.domain import calculation_registry
+from core.tests import test_utils
+from extensions.answer_summarizers import models
+
+
+class CalculationRegistryTests(test_utils.GenericTestBase):
+    """Provides testing of the calculation registry."""
+
+    def test_refresh_registry(self):
+        for name, clazz in inspect.getmembers(models, inspect.isclass):
```

Comment:
Tests should test interface, not implementation. I think this tests mimics the implementation too much and won't detect errors if the implementation is wrong.

So, try to find an independent check of correctness. One example might be verifying the number of keys, or the list of keys in the registry against a hardcoded list. It's fine if this test has to be updated each time a new calculation is added -- that happens relatively infrequently, and would prevent errors where various classes get added or excluded by mistake.